applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -238,6 +252,25 @@ def InitializeSolutionStep(self):
             # Initialize the solver current step
             self._GetSolutionStrategy().InitializeSolutionStep()
 
+    def SolveSolutionStep(self):
+        super().SolveSolutionStep()
+
+        if (self._levelset_dt_factor < 1.0):
+            # Perform the level-set convection to complete the solution step
+            if self._bfecc_convection:
+                self._GetLevelSetConvectionProcess().CopyScalarVarToPreviousTimeStep(
+                    self.main_model_part,
+                    KratosMultiphysics.DISTANCE)
+
+                self._GetLevelSetConvectionProcess().BFECCconvectPartially(
+                    self.main_model_part,
+                    KratosMultiphysics.DISTANCE,
+                    KratosMultiphysics.VELOCITY,
+                    self.settings["bfecc_number_substeps"].GetInt(),
+                    1.0 - self._levelset_dt_factor)
+            else:
+                self._GetLevelSetConvectionProcess().ExecutePartially(1.0 - self._levelset_dt_factor)
```

Comment:
In order to be consistent with the "first" level set transport (the one in the `InitializeSolutionStep` I think we should do this in the `FinalizeSolutionStep`.

As an alternative, we could do everything in the `SolveSolutionStep`. However, this would imply that the strategy predict is performed with the previous level set (not sure about the implications that this might have).

Reply:
I think the main concern is about "CopyScalarVarToPreviousTimeStep". Could you please comment about the location of Copying the new variables to the Old ones in the present strategy? I mean copying current VARIABLE to (VARIABLE, 1).

Reply:
I think that I don't get your point at all... The historical database wouldn't change (aside of the levelset convection operations) until the next clone of the modelpart, which is done in the base solver `InitializeSolutionStep` call.

Reply:
You answered my question; no change happens to the historical variables outside of the corresponding processes until (at next time-step) the next call of "InitializeSolutionStep".

So, I think I can move this part of the code to "FinalizeSolutionStep".

Reply:
Cool. This makes more sense to me.